### PR TITLE
feat(Core/AuctionHouse): Implement Auction House getAll scan

### PR DESCRIPTION
**Pull request by jerryshell**
This is a simple but effective implementation.
More scenarios may need to be considered in the future, such as player scan interval control, but I'd like to submit a small commit first.

## Changes Proposed:

This PR proposes changes to:
-  [x] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/5068

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1. I've tested it on my local server using [TradeSkillMaster](https://github.com/Wrath-AddOns/TradeSkillMasterPlus) (with an auction house of 25k items)

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/AuctionHouse/AuctionHouseMgr.cpp [hunk 1]
```diff
@@ -733,8 +733,27 @@ void AuctionHouseObject::BuildListOwnerItems(WorldPacket& data, Player* player,
 bool AuctionHouseObject::BuildListAuctionItems(WorldPacket& data, Player* player,
         std::wstring const& wsearchedname, uint32 listfrom, uint8 levelmin, uint8 levelmax, uint8 usable,
         uint32 inventoryType, uint32 itemClass, uint32 itemSubClass, uint32 quality,
-        uint32& count, uint32& totalcount, uint8 /*getAll*/, AuctionSortOrderVector const& sortOrder, Milliseconds searchTimeout)
+        uint32& count, uint32& totalcount, uint8 getAll, AuctionSortOrderVector const& sortOrder, Milliseconds searchTimeout)
```
Comment (elthehablo) [inline src/server/game/AuctionHouse/AuctionHouseMgr.cpp:736]: What is the thought process behind it being a uint8 and not a bool? I believe we should change it

**Review by elthehablo — CHANGES_REQUESTED** [review]

**Comment by Takenbacon** [comment]
Should keep in mind 3.x client network protocol has a hard limit on packet size, auctionhouse getAll is the only packet from my memory that has the potential to exceed it if uncapped, someone would need to do the math but if I recall right it's around 55-60k items.

**Comment by walkline** [comment]
Since the current auction implementation has race conditions, I would expect requests like this to make the situation worse.

**Comment by walkline** [comment]
> Since the current auction implementation has race conditions, I would expect requests like this to make the situation worse.

More details here - https://github.com/azerothcore/azerothcore-wotlk/issues/20538

**Comment by natugrumu** [comment]
I cannot begin to express how much I am looking forward to this being implemented.

**Comment by jerryshell** [comment]
https://github.com/azerothcore/azerothcore-wotlk/pull/20830

**Pull request closed without merging**